pkg/graph.go

```diff
@@ -0,0 +1,69 @@
+package pkg
+
+import "log"
+
+// resolve finds the order of dependencies for a graph
+// of nodes.
+// Inspired by algorithm from
+// https://www.electricmonk.nl/log/2008/08/07/dependency-resolving-algorithm/
+func resolve(node *Node, resolved, unresolved *Graph) {
```

Comment:
i don't understand this method signature, is it mutating `unresolved` into `resolved`? vice versa? Why not have an error return instead of a panic?

Reply:
You can read the linked blog post to see how the two slices work together - the unresolved is part of circular error detection.

Finding an error doesn't stop the whole stack and unwind, which is what we need here. Perhaps it could be done with something like a channel.